app/models/concerns/cable_ready/updatable/collections_registry.rb

```diff
@@ -12,23 +23,35 @@ def register(collection)
       end
 
       def broadcast_for!(model, operation)
-        @registered_collections.select { |c| c[:options][:on].include?(operation) }
+        @registered_collections.select { |c| c.options[:on].include?(operation) }
           .each do |collection|
           resource = find_resource_for_update(collection, model)
           next if resource.nil?
 
-          collection[:klass].cable_ready_update_collection(resource, collection[:name], model) if collection[:options][:if].call(resource)
+          collection.klass.cable_ready_update_collection(resource, collection.name, model) if collection.options[:if].call(resource)
         end
       end
 
       private
 
       def find_resource_for_update(collection, model)
-        raise ArgumentError, "Could not find inverse_of for #{collection[:name]}" unless collection[:inverse_association]
+        collection.reflection ||= collection.klass.reflect_on_association(collection.name)
+
+        collection.foreign_key ||= collection.reflection.foreign_key
```

Comment:
We use safe navigation below on `collection.reflection`. Should we do the same here?